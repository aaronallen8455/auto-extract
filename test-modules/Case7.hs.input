module Case7 where

test :: Int
test =
  let y = 5
      x = 3
   in EXTRACT@a (let z = 10 in test + EXTRACT@b (x + y * z))
