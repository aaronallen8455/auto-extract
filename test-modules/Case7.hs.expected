module Case7 where

test :: Int
test =
  let y = 5
      x = 3
   in a x y

b :: Num a => a -> a -> a -> a
b x y z = x + y * z

a :: Int -> Int -> Int
a x y = let z = 10 in test + b x y z
